```
Jun 05 04:15:47 /build/src/Storages/S3Queue/S3QueueOrderedFileMetadata.cpp:374:16: error: unused local variable 'node_metadata_str' of type 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char>>') [bugprone-unused-local-non-trivial-variable,-warnings-as-errors]
Jun 05 04:15:47   374 |     const auto node_metadata_str = node_metadata.toString();
Jun 05 04:15:47       |                ^
Jun 05 04:15:47 34132 warnings generated.
Jun 05 04:15:47 Suppressed 34678 warnings (34131 in non-user code, 547 NOLINT).
Jun 05 04:15:47 Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
Jun 05 04:15:47 1 warning treated as error
```

--- a/src/Storages/S3Queue/S3QueueMetadata.cpp
+++ b/src/Storages/S3Queue/S3QueueMetadata.cpp
@@ -167,7 +167,7 @@ S3QueueMetadata::FileMetadataPtr S3QueueMetadata::getFileMetadata(
     S3QueueOrderedFileMetadata::BucketInfoPtr bucket_info)
 {
     auto file_status = local_file_statuses->get(path, /* create */true);
-    switch (settings.mode)
+    switch (settings.mode.value)
     {
         case S3QueueMode::ORDERED:
             return std::make_shared<S3QueueOrderedFileMetadata>(

--- a/src/Storages/S3Queue/S3QueueOrderedFileMetadata.cpp
+++ b/src/Storages/S3Queue/S3QueueOrderedFileMetadata.cpp
@@ -371,7 +371,6 @@ void S3QueueOrderedFileMetadata::setProcessedImpl()
     };
 
     const auto zk_client = getZooKeeper();
-    const auto node_metadata_str = node_metadata.toString();
     std::string failure_reason;
 
     while (true)

